src/java/org/apache/cassandra/cql3/statements/AlterKeyspaceStatement.java

```diff
@@ -96,7 +98,35 @@ private void warnIfIncreasingRF(KeyspaceMetadata ksm, KeyspaceParams params)
                                                                                                         StorageService.instance.getTokenMetadata(),
                                                                                                         DatabaseDescriptor.getEndpointSnitch(),
                                                                                                         params.replication.options);
-        if (newStrategy.getReplicationFactor() > oldStrategy.getReplicationFactor())
+
+        validateTransientReplication(oldStrategy, newStrategy);
+        warnIfIncreasingRF(oldStrategy, newStrategy);
+    }
+
+    private void validateTransientReplication(AbstractReplicationStrategy oldStrategy, AbstractReplicationStrategy newStrategy)
+    {
+        if (oldStrategy.getReplicationFactor().trans == 0 && newStrategy.getReplicationFactor().trans > 0)
+        {
+            Keyspace ks = Keyspace.open(keyspace());
+            for (ColumnFamilyStore cfs: ks.getColumnFamilyStores())
+            {
+                if (cfs.viewManager.hasViews())
+                {
+                    throw new ConfigurationException("Cannot use transient replication on keyspaces using materialized views");
+                }
+
+                if (cfs.indexManager.hasIndexes())
+                {
+                    throw new ConfigurationException("Cannot use transient replication on keyspaces using secondary indexes");
+                }
+            }
+
+        }
+    }
+
+    private void warnIfIncreasingRF(AbstractReplicationStrategy oldStrategy, AbstractReplicationStrategy newStrategy)
+    {
+        if (newStrategy.getReplicationFactor().full > oldStrategy.getReplicationFactor().full)
```

Comment:
I think full is what we want here. Increasing the number of full replicas would require a repair. If the total rf is increased, but the number of full replicas doesn't change, then you're just adding transient replicas, which shouldn't require a repair or any other operator action.

Reply:
Huh, so because your quorum size increases you don't need to run even an incremental repair? Is that always true. Have to think about it some more.

But... while it might be safe to do one increase, if you increased transient replication twice in sequence and it didn't change th quorum size could you get into trouble?

Reply:
Since replication factor changes don't really fall under any of the existing JIRAs, maybe we should disable all rf changes (except transient increases) as part of the initial refactor, and make a follow on jira to enable the other rf changes after the read and write path changes are in. Then we'll be able to reason about them in isolation, and will be better equipped to do the needed testing / tool updates.

Reply:
Sounds good.